File: daily/128.最长连续序列.js
/*
 * @lc app=leetcode.cn id=128 lang=javascript
 *
 * [128] 最长连续序列
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
  const arr = [...new Set(nums.sort((a, b) => a - b))];
  let max = 0;
  for (let i = 0; i < arr.length; ++i) {
    let count = 1;
    while (arr[i + 1] - arr[i] === 1) i++, count++;
    max = Math.max(max, count);
  }
  return max;
};
// @lc code=end

const nums = [0];
console.log(longestConsecutive(nums));

File: daily/128.最长连续序列.rs
/*
 * @lc app=leetcode.cn id=128 lang=rust
 *
 * [128] 最长连续序列
 */

// @lc code=start
use std::cmp::max;
use std::collections::HashSet;

impl Solution {
    pub fn longest_consecutive(nums: Vec<i32>) -> i32 {
        let set: HashSet<i32> = nums.into_iter().collect();
        let mut arr: Vec<i32> = set.into_iter().collect();
        let mut max_count = 0;
        arr.sort_unstable();
        for mut i in 0..arr.len() {
            let mut count = 1;
            while i + 1 < arr.len() && arr[i + 1] - arr[i] == 1 {
                i += 1;
                count += 1;
            }
            max_count = max(max_count, count);
        }
        return max_count;
    }
}
// @lc code=end

